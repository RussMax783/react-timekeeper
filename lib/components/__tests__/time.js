'use strict';

var _enzyme = require('enzyme');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Time = require('../Time');

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HOUR_SELECT_BUTTON = '.react-timekeeper__hour-select';
var MINUTE_SELECT_BUTTON = '.react-timekeeper__minute-select';
var MERIDIEM_TOGGLE = '.react-timekeeper__meridiem-toggle';

jest.mock('../TimeDropdown', function () {
	return function (props) {
		return _react2.default.createElement('div', { className: 'time-dropdown-' + props.type });
	};
});

describe('Time component', function () {
	var changeUnit = jest.fn();

	describe('current unit is hour', function () {
		var unit = 'hour';
		var timeRendered = (0, _enzyme.mount)(_react2.default.createElement(_Time.Time, {
			config: {},
			unit: unit,
			changeUnit: changeUnit
			// props to suppress proptypes warnings
			, meridiem: 'am',
			changeMeridiem: jest.fn()
		}));

		test('minute is selected', function () {
			timeRendered.find(MINUTE_SELECT_BUTTON).simulate('click');
			expect(changeUnit).lastCalledWith('minute');
		});

		// should open TimeDropdown
		test('hour is selected', function () {
			timeRendered.find(HOUR_SELECT_BUTTON).simulate('click');
			expect(timeRendered.find('.time-dropdown-' + unit).length).toEqual(1);
		});
	});

	describe('current unit is minute', function () {
		var unit = 'minute';
		var timeRendered = (0, _enzyme.mount)(_react2.default.createElement(_Time.Time, {
			config: {},
			unit: unit,
			changeUnit: changeUnit
			// props to suppress proptypes warnings
			, meridiem: 'am',
			changeMeridiem: jest.fn()
		}));

		test('hour is selected', function () {
			timeRendered.find(HOUR_SELECT_BUTTON).simulate('click');
			expect(changeUnit).lastCalledWith('hour');
		});

		// should open TimeDropdown
		test('minute is selected', function () {
			timeRendered.find(MINUTE_SELECT_BUTTON).simulate('click');
			expect(timeRendered.find('.time-dropdown-' + unit).length).toEqual(1);
		});
	});

	test('change meridiem', function () {
		var changeMeridiem = jest.fn();
		var timeRendered = (0, _enzyme.shallow)(_react2.default.createElement(_Time.Time, {
			config: {},
			meridiem: 'am',
			changeMeridiem: changeMeridiem
			// props to suppress proptypes warnings
			, unit: 'hour',
			changeUnit: jest.fn()
		}));
		timeRendered.find(MERIDIEM_TOGGLE).simulate('click');
		expect(changeMeridiem).lastCalledWith('pm');

		timeRendered = (0, _enzyme.shallow)(_react2.default.createElement(_Time.Time, {
			config: {},
			meridiem: 'pm',
			changeMeridiem: changeMeridiem
			// props to suppress proptypes warnings
			, unit: 'hour',
			changeUnit: jest.fn()
		}));
		timeRendered.find(MERIDIEM_TOGGLE).simulate('click');
		expect(changeMeridiem).lastCalledWith('am');
	});

	test('should render correctly', function () {
		var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_Time.Time, {
			config: {},
			meridiem: 'pm',
			changeMeridiem: jest.fn(),
			unit: 'hour',
			changeUnit: jest.fn()
		})).toJSON();
		expect(tree).toMatchSnapshot();
	});
});