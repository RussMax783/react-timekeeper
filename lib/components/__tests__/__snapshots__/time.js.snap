exports[`Time component should render correctly 1`] = `
<div
  style={
    Object {
      "background": undefined,
      "borderRadius": "3px 3px 0 0",
      "padding": "14px 16px",
    }
  }>
  <div
    style={
      Object {
        "left": 20,
        "position": "relative",
      }
    }>
    <div
      style={
        Object {
          "display": "inline-block",
          "position": "relative",
          "textAlign": "right",
          "width": "72px",
        }
      }>
      <span
        className="react-timekeeper__hour-select"
        onClick={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "cursor": "pointer",
              "display": "inline-block",
              "fontSize": "48px",
              "lineHeight": "normal",
              "userSelect": "none",
            },
            Object {
              "animation": "x 0.6s ease-out both",
              "animationName": Object {
                "__process": [Function],
                "__radiumKeyframes": true,
              },
              "color": undefined,
            },
          ]
        } />
      
    </div>
    <span
      style={
        Object {
          "color": undefined,
          "display": "inline-block",
          "fontSize": "46px",
          "fontWeight": "500",
          "lineHeight": "normal",
          "margin": "0 5px",
          "verticalAlign": "2px",
        }
      }>
      :
    </span>
    <div
      style={
        Object {
          "display": "inline-block",
          "position": "relative",
        }
      }>
      <span
        className="react-timekeeper__minute-select"
        onClick={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "cursor": "pointer",
              "display": "inline-block",
              "fontSize": "48px",
              "lineHeight": "normal",
              "userSelect": "none",
            },
            false,
          ]
        }>
        ed
      </span>
      
    </div>
    <button
      className="react-timekeeper-button-reset react-timekeeper__meridiem-toggle"
      onClick={[Function]}
      style={
        Object {
          "color": undefined,
          "display": "inline-block",
          "fontFamily": undefined,
          "fontSize": "13px",
          "marginLeft": "2px",
          "padding": "10px 8px",
          "textTransform": "uppercase",
          "verticalAlign": "1px",
        }
      }
      type="button">
      pm
    </button>
  </div>
</div>
`;
